pkg/core/servicecontext/file.go

```diff
@@ -0,0 +1,128 @@
+package servicecontext
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+// NewFile creates a new context type
+func NewFile() IContext {
+	cfg := &File{}
+
+	return cfg
+}
+
+// File is a type which describes a context file
+type File struct{}
+
+const errorFormat = "%v: %w"
+
+const envName = "RHOAS_CONTEXT"
+
+// Load loads the profiles from the context file. If the context file doesn't exist
+// it will return an empty context object.
+func (c *File) Load() (*Context, error) {
+	file, err := c.Location()
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = os.Stat(file)
+	if os.IsNotExist(err) {
+		return nil, err
+	}
+	if err != nil {
+		return nil, fmt.Errorf(errorFormat, "unable to check if context file exists", err)
+	}
+	// #nosec G304
+	data, err := ioutil.ReadFile(file)
+	if err != nil {
+		return nil, fmt.Errorf(errorFormat, "unable to read context file", err)
+	}
+	var cfg Context
+	err = json.Unmarshal(data, &cfg)
+	if err != nil {
+		return nil, fmt.Errorf(errorFormat, "unable to parse contexts", err)
+	}
+	return &cfg, nil
+}
+
+// Save saves the given profiles to the context file.
+func (c *File) Save(cfg *Context) error {
+	file, err := c.Location()
+	if err != nil {
+		return err
+	}
+	data, err := json.MarshalIndent(cfg, "", "  ")
+	if err != nil {
+		return fmt.Errorf("%v: %w", "unable to marshal context", err)
+	}
+	rhoasCfgDir, err := DefaultDir()
+	if err != nil {
+		return err
+	}
+	if _, err = os.Stat(rhoasCfgDir); os.IsNotExist(err) {
+		err = os.Mkdir(rhoasCfgDir, 0o700)
+		if err != nil {
+			return err
+		}
+	}
+	err = ioutil.WriteFile(file, data, 0o600)
+	if err != nil {
+		return fmt.Errorf(errorFormat, "unable to save context", err)
+	}
+	return nil
+}
+
+// Remove removes the context file.
+func (c *File) Remove() error {
+	file, err := c.Location()
+	if err != nil {
+		return err
+	}
+	_, err = os.Stat(file)
+	if os.IsNotExist(err) {
+		return nil
+	}
+	err = os.Remove(file)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Location gets the path to the context file
+func (c *File) Location() (path string, err error) {
+
+	if rhoasContext := os.Getenv(envName); rhoasContext != "" {
+		path = rhoasContext
+	} else {
+		rhoasCtxDir, err := DefaultDir()
+		if err != nil {
+			return "", err
+		}
+		path = filepath.Join(rhoasCtxDir, "contexts.json")
```

Comment:
When we will be wanting to export/import will we want to support YAML and JSON? Since the context file will be editable by users they might want YAML as you can comment etc. Something to consider.

Reply:
I think we going to do it by output format/command output.

Actual underlaying storage format needs to be consistent/singular and yaml will be odd choice.

Reply:
Yep agreed.

Reply:
Actual migration of config to yaml would not be a problem - if we feel that yaml will be better format for config and context we can use it. I'm neutral

Reply:
Actually the key difference with this file is it is user editable/visible.
I think users like choice in these matters, and some users might want a YAML file in their local context. Just something to consider.

Reply:
Sounds reasonable. @rkpattnaik780 have final choice to what format we will use. 
Let's leave config out of it. If we want to migrate it to yaml that should be separate PR etc.